File: 2020_year/12_month/2_week/jihun/Algo/2231_runtime.js
let input = require('fs').readFileSync('./dev/stdin').toString().trim().split(' ');

let data = Number(input)

const recursive = function(current, depth) {
    for (let i = 0; i < 10; i++) {
        let targetNum
        if (depth !== 0) {
            targetNum = current.toString() + i.toString()
        } else {
            targetNum = i.toString()
        }
        let targetSum = targetNum.split('').map(target => Number(target)).reduce((acc, curr) => (acc + curr))
        
        if (Number(targetNum) + targetSum === data) {
            return Number(targetNum)
        } else if (targetNum > data) {
            return 0
        }
    }
    return recursive(current+1, depth+1)
}

console.log(recursive(0, 0))

File: 2020_year/12_month/2_week/jihun/Algo/2231_runtime.py
import sys
sys.stdin = open('./dev/stdin')

data = int(input())

def recursive(current, depth):
    for i in range(10):
        if (depth != 0):
            target_num = '{}{}'.format(current, i)
        else:
            target_num = '{}'.format(i)
        
        target_sum = 0
        for part in target_num:
            target_sum += int(part)
        
        if (int(target_num) + target_sum == data):
            return int(target_num)
        elif (int(target_num) > data):
            return 0
    return recursive(current+1, depth+1)


print(recursive(0,0))